### PR TITLE
Organization dropdown bug fix

### PR DESCRIPTION
**Pull request by mwodahl**
# PR Details
## Description

This PR modifies the organization dropdown menu to update correctly after a user-organization assignment is created, deleted, or edited for the current user. Additionally, the tabs displayed now update correctly after a new organization is selected from the dropdown menu. 

## How Has This Been Tested?

This has been tested locally via docker-compose with user-organization add, edit, and delete. Additionally, organization delete has been tested and the tabs shown to the user have been verified to update correctly after switching organizations.

## Types of changes

- [ x ] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] My changes require new environment variables:
  - [ ] I have updated the docker-compose, K8s YAML, and all dependent deployment configuration files.
- [ ] My changes require updates to the documentation:
  - [ ] I have updated the documentation accordingly.
- [ x ] My changes require updates and/or additions to the unit tests:
  - [ x ] I have modified/added tests to cover my changes.
- [ x ] All existing tests pass.

### CODE REVIEW

File: webapp/src/features/adminOrganizationTab/AdminOrganizationTab.tsx [hunk 1]
```diff
@@ -59,8 +61,19 @@ const AdminOrganizationTab = () => {
...
+
   useEffect(() => {
-    dispatch(getOrgData({ orgName: 'all', all: true, specifiedOrg: undefined }))
+    dispatch(getOrgData({ orgName: 'all', all: true, specifiedOrg: undefined })).then(() => {
+      if (defaultOrgData) {
+        const selectedOrg = (orgData ?? []).find(
+          (organization: AdminOrgSummary) => organization?.name === defaultOrgName
+        )
+        dispatch(setSelectedOrg(selectedOrg))
+        defaultOrgData = null
```
Comment (dmccoystephenson) [inline webapp/src/features/adminOrganizationTab/AdminOrganizationTab.tsx:76]: A comment explaining why the defaultOrgData variable is being set to null here and what effect that has would be helpful.

**Review by dmccoystephenson — CHANGES_REQUESTED** [review]

**Review by drewjj — APPROVED** [review]
This is actually really nice. Everything seems to work well when switching between orgs with enforcing role switches. The dropdown definitely gets updated properly with the adding or removal of an org. 

Looks like the one thing that is a little janky is the RSU map doesn't refresh properly. Not sure if this was intended to be just the dropdown or if it intended to get the org switching working entirely.

File: webapp/src/features/adminOrganizationTab/AdminOrganizationTab.tsx (hunk 1 above)
Comment (mwodahl) [reply webapp/src/features/adminOrganizationTab/AdminOrganizationTab.tsx:76]: Added a comment to explain this useEffect logic.

Comment (dmccoystephenson) [reply webapp/src/features/adminOrganizationTab/AdminOrganizationTab.tsx:76]: Awesome!

**Review by dmccoystephenson — APPROVED** [review]
Looks good to me!

**Pull request merged**